cpp/src/arrow/compute/kernels/scalar_round.cc

```diff
@@ -452,6 +468,127 @@ struct Round<ArrowType, kRoundMode, enable_if_decimal<ArrowType>> {
   }
 };
 
+template <typename ArrowType, RoundMode RndMode, typename Enable = void>
+struct RoundBinary {
+  using CType = typename TypeTraits<ArrowType>::CType;
+  using State = RoundOptionsWrapper<RoundBinaryOptions>;
+
+  explicit RoundBinary(const State& state, const DataType& out_ty) {}
+
+  template <typename T = ArrowType, typename CType0 = typename TypeTraits<T>::CType0,
+            typename CType1 = typename TypeTraits<T>::CType1>
+  enable_if_floating_value<CType> Call(KernelContext* ctx, CType0 arg0, CType1 arg1,
+                                       Status* st) const {
+    // Do not process Inf or NaN because they will trigger the overflow error at end of
+    // function.
```

Comment:
Do you have any tests with infinite or NaN?

Reply:
Yes, they are in TEST_F(TestBinaryRoundFloating, Round) around line 1161.